src/lifecycle.js

```diff
@@ -111,7 +111,7 @@ function addAttributeToPropertyLinks (target, component) {
   }
 
   for (var attribute in componentAttributes) {
-    if (hasOwn(componentAttributes, attribute) && !hasOwn(target, attribute)) {
+    if (hasOwn(componentAttributes, attribute) && !hasOwn(target, camelCase(attribute))) {
```

Comment:
I agree in that we've opened up the of a bug since the knowledge that the property is camel cased is now spread in two functions.